Convey glossary (#2158)
giantswarm/docs: src/content/meta/glossary.md

---
title: Glossary
description: A page with the term definitions Giant Swarm uses in the documentation.
search: false
owner:
  - https://github.com/orgs/giantswarm/teams/sig-docs
---

> We used a set of terms across our documentation that could be no clear to readers. In the following article we define of those terms to clarify the complexity that could carry.

Reviewer: ```suggestion
We use a set of terms across our documentation that could be unclear to readers. In the following article we define those terms to clarify their meaning in the context of Giant Swarm.
```

## Cloud Native Developer Platform

A Cloud Native Developer Platform is an integrated collection of capabilities exposed by intuitive interface that facilitates the application development lifecycle. It offers a flexible configuration to build a system that encodes your company practices. It integrates with different open source solutions to enable automation, observability and security for your workloads.

## Smart Platform Engineering

Smart Platform Engineering refers to the design and development of intelligent platforms that help you to create the desired Developer Platform for your teams in a quick fashion. Instead of starting from scratch to build piece by piece an entire platform meanwhile you operate it 24/7, leverage in our knowledge, product and support to reach your goals in a decent amount of time.

## Platform API

Is the entrypoint of the Cloud Native Developer Platform. Everything in the platform is exposed via the API, allowing the automation of processes and fostering standardization. The Platform API is a fully compliant Kubernetes API extended with some custom resources to enable a variety of use cases.

## Management Cluster

> It is a Kubernetes cluster that acts a central point of the platform. It is in charge of host platform wide services like observability or security to ensure visualization and compliance over all the workloads in all clusters.

Reviewer: ```suggestion
It is a Kubernetes cluster that acts as a central point of the platform. It is in charge of hosting platform wide services like observability or security and to ensure visualization and compliance over all the workloads in all workload clusters.
```

Reviewer: Theoretically it is not central and all WCs, but per region/DC, except in the cases where we run multi-provider like Smart Factory edge.

> ---

Author: it is a meta page so it will be in -> https://docs.giantswarm.io/meta/ (and reference from different doc entries, but not in the sidebar menu) If we don't change it

search: false
owner:
  - https://github.com/orgs/giantswarm/teams/sig-docs
last_review_date: 2024-03-08
---

We use a set of terms across our documentation that could be unclear to readers. In the following article we define those terms to clarify their meaning in the context of Giant Swarm.

## Cloud Native Developer Platform

A Cloud Native Developer Platform is an integrated collection of capabilities exposed through intuitive interfaces that facilitate the application development lifecycle. It offers flexible configuration to build a system that encodes your company practices. It integrates with different open source solutions to enable automation, connectivity, observability, and security for your workloads.

## Smart Platform Engineering

Smart Platform Engineering refers to the design and development of intelligent platforms that help you create the desired Developer Platform for your teams in a quick fashion. Instead of starting from scratch building an entire platform piece by piece while you operate it 24/7, you can leverage our knowledge, product, and support to reach your goals in a much shorter amount of time.

## Platform API

> Is the entrypoint of the Cloud Native Developer Platform. Everything in the platform is exposed via the API, allowing the automation of processes and fostering standardization. The Platform API is a fully compliant Kubernetes API extended with some custom resources to enable a variety of use cases.

Author: What about pointing to the Platform API intro page?

> Is the entrypoint of the Cloud Native Developer Platform. Everything in the platform is exposed via the API, allowing the automation of processes and fostering standardization. The Platform API is a fully compliant Kubernetes API extended with some custom resources to enable a variety of use cases.

Author: I will do that once we have the entry there :D

---
title: Glossary
description: A page with the term definitions Giant Swarm uses in the documentation.
search: false
owner:
  - https://github.com/orgs/giantswarm/teams/sig-docs
user_questions:
  - "What is a cloud-native developer platform?"
  - "What is a developer environment?"
  - "What is a management cluster?"
  - "What is a managed app?"
  - "What is the platform API?"
  - "What is a platform operator?"
  - "What is smart platform engineering?"
  - "What is a workload cluster?"
---

> We use a set of terms across our documentation that could be unclear to readers. In the following article we define those terms to clarify their meaning in the context of Giant Swarm.

Reviewer 2: [super minor] From a technical writing perspective, I would just throw this sentence away. No introduction is needed, and we also probably shouldn't mention "readers" but rather "you" (you = reader).

## Cloud native developer platform

> A cloud-native developer platform is an integrated collection of capabilities exposed through intuitive interfaces that facilitate the application development lifecycle. It offers flexible configuration to build a system that encodes your company practices. It integrates with different open source solutions to enable automation, connectivity, observability, and security for your workloads.

Reviewer 2: Should we mention the more commonly known term "Internal Developer Platform (IDP)" as well? It basically means the same, but is specifically meant for _within_/_across_ one company or organization.

Author: yeah good point

## Developer environment

> In our context, a developer environment is the entity that defines the space where developers will deploy their applications. It can be mapped to a Kubernetes namespace or cluster. Platform capabilities like access management, security, or observability are related to those environments.

Author: ```suggestion
In our context, a developer environment defines the space where developers will deploy their applications. It can be mapped to a particular namespace or cluster. Platform capabilities like access management, security, or observability are related to those environments.
```

> In our context, a developer environment is the entity that defines the space where the developers will deploy their applications. It can be mapped to a Kubernetes namespace(s) or a cluster. The Platform capabilities like access management, security or observability are related to those environments.

Author: ```suggestion
In our context, a developer environment is the entity that defines the space where developers will deploy their applications. It can be mapped to a Kubernetes namespace or cluster. Platform capabilities like access management, security, or observability are related to those environments.
```

## Management cluster

> It is a Kubernetes cluster that acts a central point of the platform. It is in charge of hosting platform wide services like observability or security to ensure visualization and compliance over all the workloads in all clusters. [Platform operator](#platform-operator) have access to the its [API](#platform-api) to manage [workload clusters](#workload-cluster) and [managed apps](#managed-apps).

Reviewer 3: A few issues:
* the sentence `It acts a central point` should be `It acts as a central point.`;
* the phrase `the its API` should be `its API.`;
* the phrase `Platform operator have access` should be `Platform operators have access`.

> It is a Kubernetes cluster that acts a central point of the platform. It is in charge of hosting platform wide services like observability or security to ensure visualization and compliance over all the workloads in all clusters. [Platform operator](#platform-operator) have access to the its [API](#platform-api) to manage [workload clusters](#workload-cluster) and [managed apps](#managed-apps).

Reviewer 4: ```suggestion
A Management Cluster is a Kubernetes cluster that acts as a central management point of the platform. This cluster hosts platform wide services for gitops, observability, security and user management to ensure automation and visibility of the platform and workloads in all clusters. Users can create new clusters via a Management Cluster. [Platform engineers](#platform-engineers) usually have access to the [API](#platform-api) of the Management Cluster to manage [workload clusters](#workload-cluster) and services of the platform.
```

Reviewer 2: As a user, I wouldn't understand

> automation and visibility of the platform and workloads in all clusters

What is made visible? What is automated? Maybe this can be a bullet point list. Very roughly like this:

- Users can create new clusters or apps through the API of the Management Cluster. In the same way, GitOps can be used to create and manage clusters and apps.
- Users can view installed apps and deployed workload clusters and their state through the API of the Management Cluster.
- The operators running on the Management Clusters manage the workload clusters and apps.
- (...something about observability and Grafana, maybe...)

Also `s/gitops/GitOps/`

Author: I have tried to simplify it. I would not define the whole concept here, it is just a glossary and IMO we can point to intro page for MC

> ## Managed app

Reviewer 4: We decided quite some time ago to stop using the term "Managed app" and think more in terms of bundles/capabilities. @puja108 what do you think? How shall we call this here? "Managed services", "Managed capabilities", "Managed bundles"?

Reviewer 2: Then what about the `App` resource which we'll have to describe? I think the term "app" in general is well-understood by users, so maybe we can combine the ideas somehow when we explain it (or make up a new heading/term).

Author: I am not against introducing a new concept but in the end, the app which will carry the capability or service will be there and we will manage it.

Reviewer: Yes, we wanted to get rid of the managed app concept as it puts too much focus on apps and distracts from the actual capabilities they should enable. Among others, we wanted to avoid tech comparisons based on apps.

I prefer to use "managed capabilities" as services and bundles still have technical connotations.

That said, this does not mean that the App CR is not used here right now. It should, however, be kept in mind that that resource might go away, e.g. be replaced with something upstream like HelmRelease CR. The app in our case is an implementation detail of how we deploy helm charts declaratively. It can be defined separately, but then we're on an API level, not on a concept level like here.

I can do a suggestion that reflects the change here.

> ## Managed app

Reviewer 2: All entries should strictly be sorted alphabetically (one minor such difference is left: `"Managed apps" < "Management cluster"`). Can we also add a `<!-- All headings must be sorted alphabetically -->` at the top, just below the YAML header, so any editors take care of this?

> The managed apps are a set of curated diverse components that can be used in the [platform](#cloud-native-developer-platform) to enable capabilities across the [developer environments](#developer-environment). As an example, in case the developers want to allocate dynamically DNS records for their services they can leverage in [external-dns](https://github.com/giantswarm/external-dns-app/). The managed apps are operated by us and we provide regular upgrades to avoid security vulnerabilities and bugs.

Reviewer 2: > curated diverse

🤔 

```suggestion
The managed apps are a set of curated components that can optionally be used in the [platform](#cloud-native-developer-platform) to enable diverse capabilities across the [developer environments](#developer-environment). As an example, in case the developers want to dynamically allocate DNS records for their services, they can leverage [external-dns](https://github.com/giantswarm/external-dns-app/). The managed apps are operated by us and we provide regular upgrades to avoid security vulnerabilities and bugs.
```

Also "operated by us" is something we should clarify now or later.

## Platform API

> Is the entrypoint of the [cloud-native developer platform](#cloud-native-developer-platform). Everything in the platform is exposed via the API, allowing the automation of processes and fostering standardization. The Platform API is a fully compliant Kubernetes API extended with some custom resources to enable a variety of use cases. [Know more here](#future-link-to-platform-api-intro-page).

Reviewer 4: We can also add a link to upstream CRD docs.

> Is the entrypoint of the [cloud-native developer platform](#cloud-native-developer-platform). Everything in the platform is exposed via the API, allowing the automation of processes and fostering standardization. The Platform API is a fully compliant Kubernetes API extended with some custom resources to enable a variety of use cases. [Know more here](#future-link-to-platform-api-intro-page).

Reviewer 2: > The Platform API is a fully compliant Kubernetes API extended with some custom resources to enable a variety of use cases.

This may unwittingly sound as if it was a custom-built, proprietary solution. What about this (includes Timo's suggestions, but avoiding the word "our"):

```suggestion
The Platform API is the entrypoint of Giant Swarm's [cloud-native developer platform](#cloud-native-developer-platform). Everything in the platform is exposed via the Platform API, allowing the automation of processes, including GitOps support, and fostering standardization. The Platform API is the Kubernetes API of the Management Cluster, extended with custom resources such as [`App`](#managed-apps] to enable a variety of use cases. [Learn more here](#future-link-to-platform-api-intro-page).
```

> ## Platform operator

Reviewer 4: Let's call this "Platform team" or "Platform engineer". I'd prefer the first. "Operator" can be really confusing since it is also a software component.

> The platform operator is the person or team that is in charge of the operation of the [cloud-native developer platform](#cloud-native-developer-platform). They are responsible for the management of the [management cluster](#management-cluster) and its capabilities.

Reviewer 4: ```suggestion
The platform team is a team that is in charge of providing the [cloud-native developer platform](#cloud-native-developer-platform). With the help of Giant Swarms managed platform they compose, manage and customize the platform for the specific needs of the development teams within their company.
```

> The platform team is in charge of providing the [cloud-native developer platform](#cloud-native-developer-platform). With the help of the Giant Swarms managed platform, they compose, manage and customize the platform for the specific needs of the development teams within their company.

Author: ```suggestion
The platform team is in charge of providing a [cloud-native developer platform](#cloud-native-developer-platform). With the help of the Giant Swarms managed platform, they compose, manage and customize the platform for the specific needs of the development teams within their company.
```

> The platform team is a team that is in charge of providing the [cloud-native developer platform](#cloud-native-developer-platform). With the help of Giant Swarms managed platform they compose, manage and customize the platform for the specific needs of the development teams within their company.

Author: ```suggestion
The platform team is in charge of providing the [cloud-native developer platform](#cloud-native-developer-platform). With the help of the Giant Swarms managed platform, they compose, manage and customize the platform for the specific needs of the development teams within their company.
```

## Smart platform engineering

> Smart Platform Engineering refers to the design and development of intelligent platforms that help you to create the desired Developer Platform for your teams in a quick fashion. Instead of starting from scratch to build piece by piece an entire platform meanwhile you operate it 24/7, leverage in our knowledge, product and support to reach your goals in a decent amount of time.

Reviewer 4: ```suggestion
Smart Platform Engineering is the design and development of intelligent platforms that help create a developer platform for your teams. Instead of starting from scratch and building an entire platform piece by piece while running it 24/7, use our knowledge, products and support to achieve your goals faster.
```

## Workload cluster

> The customer platform teams can define several clusters where the developers can run their applications based on different requirements like location, stage or isolation.

Reviewer 2: Let's give examples here, since "stage" may not be very clear.

I don't have a concrete wording in mind, but let me list what I mean: `{dev,staging,prod}-{apac,emea,eu,us,...}-{chatbot,payments,shop,...}`.

> Smart Platform Engineering is the design and development of intelligent platforms that help create a developer platform for your teams. Instead of starting from scratch and building an entire platform piece by piece while running it 24/7, use our knowledge, products and support to achieve your goals faster.

Author: ```suggestion
Smart Platform Engineering is designing and developing intelligent platforms that help create a developer platform for your teams. Instead of starting from scratch and building an entire platform piece by piece while running it every day, use our knowledge, products and support to achieve your goals faster.
```

> The customer platform teams can define several clusters where the developers can run their applications based on different requirements like location (Ireland, US, China), stage (dev, staging, production) or isolation (payments, ecommerce).

Author: ```suggestion
The customer platform teams can define several clusters where the developers can run their applications based on different requirements, such as location (Ireland, US, China), stage (development, staging, production), or isolation (payments, e-commerce).
```
